hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -545,12 +551,22 @@ public void onPreHandle(@NonNull final Event event, @NonNull final HederaState s
         daggerApp.preHandleWorkflow().preHandle(readableStoreFactory, creator.accountId(), transactions.stream());
     }
 
-    public void onNewRecoveredState() {
-        // (FUTURE) - Consider supporting an option to dump state at the end of event stream replay;
-        // but always close the block manager so replay will end with a complete record file
+    public void onNewRecoveredState(@NonNull final MerkleHederaState recoveredState) {
+        try {
+            if (shouldDump(daggerApp.initTrigger(), MOD_POST_EVENT_STREAM_REPLAY)) {
+                dumpModChildrenFrom(recoveredState, MOD_POST_EVENT_STREAM_REPLAY, MerkleStateChild.childrenToDump());
+            }
+        } catch (Exception e) {
+            logger.error("Error dumping state after migration at MOD_POST_EVENT_STREAM_REPLAY", e);
+        }
+        // Always close the block manager so replay will end with a complete record file
         daggerApp.blockRecordManager().close();
     }
```

Comment:
What is the tradeoff here? What does it mean to dump state? What is a DumpCheckpoint? What is the effect on this for restart, upgrade, etc? Where is this design documented? I always thought the state dumpers were temporary for just the mono->mod migration. Are they to be kept?

Reply:
We wanted to still keep StateDumper code, so we can use them when needed. For example, with this change we validated the record streams and the state is the same for a four 30 min intervals and validated with `release/0.49` branch, to make sure nothing is broken from the initial mod-service code.